### PR TITLE
Allow players to select their role from "bag"

### PR DESCRIPTION
**Pull request by vermentor**

### CODE REVIEW

File: client/src/PlayerLanding/PlayerRoleSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <Heading>Select a Role</Heading>
+      <Grid columns="3" gap="3" width="auto">
+        {Object.entries(game.roleBag).map(([role, taken]) => (
+          <button
+            className={classNames(taken && "opacity-40")}
+            disabled={taken || isSetRoleLoading}
+            onClick={async () => {
+              if (await takeRole(role as Role)) {
+                setRole(player!, role as Role);
+              }
```
Comment (vermentor) [inline client/src/PlayerLanding/PlayerRoleSelect.tsx:27]: Can the websocket handle this many concurrent requests?  Will it handle these requests serially?

File: server/src/types/UnifiedGame.ts [hunk 2]
```diff
@@ -40,6 +40,7 @@ export interface BaseUnifiedGame {
   deadVotes: Record<string, boolean>
   travelers: Record<string, boolean>
   alignmentsOverrides: Record<string, Alignment>
+  roleBag: Record<Role, boolean>
```
Comment (vermentor) [inline server/src/types/UnifiedGame.ts:43]: In retrospect I don't think this should be in `game` at all

File: client/src/PlayerLanding/PlayerRoleSelect.tsx (hunk 1 above)
Comment (Alphy11) [reply client/src/PlayerLanding/PlayerRoleSelect.tsx:33]: Let's test it?

**Comment by vermentor** [comment]
Rebasing harder than rewriting, so I'm rewriting this branch against demo

**Pull request closed without merging**